autogluon/utils/tabular/ml/models/abstract/abstract_model.py

```diff
@@ -389,23 +392,16 @@ def hyperparameter_tune(self, X_train, X_test, Y_train, Y_test, scheduler_option
         hpo_model_performances = {}
         for trial in sorted(hpo_results['trial_info'].keys()):
             # TODO: ignore models which were killed early by scheduler (eg. in Hyperband). How to ID these?
-            file_id = f"trial_{trial}"  # unique identifier to files from this trial
-            file_prefix = f"{file_id}_"
-            trial_model_name = f"{self.name}_{file_id}"
-            trial_model_path = f"{self.path}{file_prefix}"
+            file_id = "trial_" + str(trial)  # unique identifier to files from this trial
+            trial_model_name = self.name + os.path.sep + file_id
+            trial_model_path = self.path_root + trial_model_name + os.path.sep
             hpo_models[trial_model_name] = trial_model_path
             hpo_model_performances[trial_model_name] = hpo_results['trial_info'][trial][scheduler._reward_attr]
 
         logger.log(15, "Time for %s model HPO: %s" % (self.name, str(hpo_results['total_time'])))
-        self.params.update(best_hp)
```

Comment:
Don't we want to update this model's params with the best_hp?  For example, if a user just calls model.hyperparameter_tune() themselves (outside of task.fit()), they presumably expect the model now to have its optimal hyperparameter values.

Reply:
We don't, the model which hyperparameter_tune is called for is never used and shouldn't be treated as a real model. The return value of HPO contains the real models.